fix: adiciona validação de status

marcospinto1988 opened the pull request:
# 🔀 Pull Request

## 📋 Descrição

Sistema aceitava qualquer texto como status.
## Solução
- Lista de 4 status válidos
- Validação antes de alterar
- Mensagens de erro claras
## Teste
1. Execute `python tickets.py`
2. Tente status inválido: rejeita
3. Tente status válido: aceita

**Issue relacionada:** #[2]

---

## 🔧 Tipo de Mudança

Marque o tipo desta mudança:

- [ x ] 🐛 Correção de bug (fix)
- [ ] ✨ Nova funcionalidade (feature)
- [ ] ♻️ Refatoração (refactor)
- [ ] 📝 Documentação (docs)
- [ ] 🧪 Testes (test)

---

## ✅ Checklist de Qualidade

Antes de solicitar revisão, verifique:

- [ x ] O código roda sem erros
- [ x ] Testei manualmente a funcionalidade
- [ ] Atualizei a documentação (README, comentários)
- [ x ] O código segue boas práticas (nomes claros, sem duplicação)
- [ x ] Revisei meu próprio código antes de submeter

---

Prioridade 

[ x ] high-priority 

[ ] medium-priority 

[ ] low-priority


labels: bug, high-priority

## 🧪 Como Testar?

Descreva os passos para testar suas mudanças:

1. Execute `python tickets.py`
2. Faça login com usuário `admin`
3. Teste [funcionalidade específica]
4. Verifique que [resultado esperado]

---

## 📸 Evidências

Se aplicável, adicione prints ou vídeos mostrando as mudanças funcionando.

---

## 💬 Observações Adicionais

Alguma informação extra que o revisor deveria saber?

---

## 👥 Revisor

@[marcar membro do grupo para revisar]

EdHungria committed:
fix: adiciona validação de status

-Lista de status válidos
-Validação antes de alterar
-Tratamento de erro

Closes #1

EdHungria committed:
fix: adiciona validação de data
- função validar_data()
- Formato DD/MM/AAAA obrigatório
- Não aceita data futura
- 3 tentativas
Closes #2

EdHungria committed:
refactor: move credenciais para arquivo de configuração
- Cria config.py para configurações centralizadas
- Remove senha hardcoded do código principal
- Adiciona suporte a múltiplos usuários
- Move STATUS_VALIDOS para configuração

Closes #3

EdHungria committed:
feat: adiciona arquivo .gitignore
- Ignora arquivos Python temporários
- Ignora ambientes virtuais
- Ignora arquivos de IDE
- Ignora arquivos do sistema operacional
- Ignora configurações locais sensíveis

Closes #4

EdHungria committed:
docs: melhora documentação do README
- Adiciona a descrição do projeto
- Documenta funcionalidades
- Inclui instruções de instalação e uso
- Descreve status válidos e formato de data
- Adiciona estrutura do projeto
- Lista correções realizadas
- Inclui informações da equipe

Closes #5

EdHungria reviewed (approved):
Bom tratamento com try/except para capturar ValueError. A função está bem integrada e o ticket só é criado com data válida. Sugestão: validar também datas futuras (if data_parsed > datetime.today(): rejeitar), dependendo dos requisitos do sistema de suporte.

marcospinto1988 commented:
Agradeço a sugestão. 
Irei revisar, agradeço a sugestão!

EdHungria reviewed (approved):
Parabéns pela correção completa do Bug #2! A validação de data está robusta e bem estruturada: função dedicada, checagens múltiplas (formato, validade real, futura, ano mínimo), loop de tentativas com feedback amigável e cancelamento seguro. O código ficou mais profissional, seguro e user-friendly.

As sugestões nos comentários são apenas refinamentos menores (constante para tentativas). Nada crítico – a implementação já resolve o problema perfeitamente.

EdHungria committed:
docs: adiciona CHANGELOG.md

- Documenta todas as mudanças do projeto
- Organizado por versão e data
- Categorias: Adicionado, Mudado, Corrigido, Segurança
- Segue padrão Keep a Changelog

marcospinto1988 reviewed (commented):
Muito obrigado pelo comentário, irei pedir sugestões futuras!

EdHungria committed:
Merge pull request #9 from EdHungria/fix/remover_senha_hardcoded

Refactor: Move credenciais para config.py

EdHungria committed:
Merge pull request #10 from EdHungria/feature/adicionar-gitignore

Feature: Adiciona .gitignore

EdHungria committed:
Merge pull request #11 from EdHungria/docs/melhorar-readme

Docs: Melhora README do projeto

EdHungria committed:
Merge pull request #13 from EdHungria/fix/validacao-data

Fix: Validação de data

EdHungria committed:
Merge pull request #15 from EdHungria/docs/criar-changelog

Docs: Adiciona CHANGELOG.md

EdHungria committed:
resolve merge main into fix/validacao-status (conflito resolvido - usa STATUS_VALIDOS do config.py)

marcospinto1988 commented on tickets.py:
```diff
@@ -27,38 +36,80 @@ def menu_principal():
-    descricao = input("Descrição detalhada: ")
-    usuario = input("Usuário solicitante: ")
-    data = input("Data (DD/MM/AAAA): ")  # BUG #2: Aceita qualquer coisa!
+    # Validação de data com 3 tentativas
```
A função no programa esta atendendo todos os critérios para um bom código, atendendo ao que foi solicitado, não havendo a necessidade de se realizar modificações.

marcospinto1988 replied:
Obrigado pelo comprometimento e nos incentivar a desenvolver um bom trabalho.

marcospinto1988 reviewed (commented):
Obrigado pelo comprometimento e nos incentivar a desenvolver um bom trabalho.
